great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -449,19 +495,87 @@ def _get_round_decimals_using_heuristics(
         )
         if round_decimals is None:
             round_decimals = MAX_DECIMALS
-        elif not isinstance(round_decimals, int) or (round_decimals < 0):
-            raise ge_exceptions.ProfilerExecutionError(
-                message=f"""The directive "round_decimals" for {self.__class__.__name__} can be 0 or a
+        else:
+            if not isinstance(round_decimals, int) or (round_decimals < 0):
+                raise ge_exceptions.ProfilerExecutionError(
+                    message=f"""The directive "round_decimals" for {self.__class__.__name__} can be 0 or a
 positive integer, or must be omitted (or set to None).
 """
-            )
-        metric_value: Number
-        if all(
-            [
-                np.issubdtype(type(metric_value), np.integer)
-                for metric_value in metric_values
-            ]
-        ):
+                )
+
+        if np.issubdtype(metric_values.dtype, np.integer):
             round_decimals = 0
 
         return round_decimals
+
+    def _get_bootstrap_estimate(
+        self,
+        metric_values: np.ndarray,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        **kwargs,
+    ) -> tuple:
+        # Obtain false_positive_rate from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        false_positive_rate: np.float64 = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=kwargs.get("false_positive_rate", 5.0e-2),
+            expected_return_type=(float, np.float64),
+            variables=variables,
+            parameters=parameters,
+        )
+        if not (0.0 <= false_positive_rate <= 1.0):
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"The confidence level for {self.__class__.__name__} is outside of [0.0, 1.0] closed interval."
+            )
+
+        # Obtain num_bootstrap_samples override from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        num_bootstrap_samples: Optional[
+            int
+        ] = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=kwargs.get("num_bootstrap_samples"),
+            expected_return_type=None,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        n_resamples: int
+        if num_bootstrap_samples is None:
+            n_resamples = DEFAULT_BOOTSTRAP_NUM_RESAMPLES
+        else:
+            n_resamples = num_bootstrap_samples
+
+        return compute_bootstrap_quantiles(
+            metric_values=metric_values,
+            false_positive_rate=false_positive_rate,
+            n_resamples=n_resamples,
+        )
+
+    def _get_deterministic_estimate(
+        self,
+        metric_values: np.ndarray,
+        domain: Domain,
+        *,
```

Comment:
Making the same note

Reply:
@cdkini It returns 2 (two) separate floats. Can we do something smart about it?